pkg/compact/compact.go

```diff
@@ -1095,7 +1124,10 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 
 	begin = time.Now()
 
-	if err := block.Upload(ctx, cg.logger, cg.bkt, bdir, cg.hashFunc); err != nil {
+	tracing.DoInSpan(ctx, "compaction_block_upload", func(ctx context.Context) {
+		err = block.Upload(ctx, cg.logger, cg.bkt, bdir, cg.hashFunc)
+	})
+	if err != nil {
```

Comment:
Let's add some spans for deleting block as well.

Reply:
Done in a recent commit.